requirements-dev.txt

```diff
@@ -2,62 +2,46 @@
-cfgv==3.1.0               # via pre-commit
-click==7.1.2              # via black, pip-tools
-decorator==4.4.2          # via ipython, traitlets
-distlib==0.3.0            # via virtualenv
-filelock==3.0.12          # via tox, virtualenv
-flake8-mypy==17.8.0       # via -r requirements-dev.in
-flake8==3.8.2             # via -r requirements-dev.in, flake8-mypy
-identify==1.4.19          # via pre-commit
-importlib-metadata==1.6.1  # via flake8, pluggy, pre-commit, pytest, virtualenv
-ipdb==0.12                # via -r requirements-dev.in
-ipython-genutils==0.2.0   # via traitlets
-ipython==7.15.0           # via ipdb
-isort==4.3.21             # via -r requirements-dev.in
-jedi==0.17.0              # via ipython
-mccabe==0.6.1             # via flake8
-more-itertools==8.3.0     # via pytest
-mypy-extensions==0.4.3    # via mypy
-mypy==0.780               # via flake8-mypy
-nodeenv==1.4.0            # via pre-commit
-packaging==20.4           # via pytest
-parso==0.7.0              # via jedi
-pathspec==0.8.0           # via black
-pexpect==4.8.0            # via ipython
-pickleshare==0.7.5        # via ipython
-pip-tools==5.2.0          # via -r requirements-dev.in
-pluggy==0.13.1            # via pytest, tox
-pre-commit==1.17.0        # via -r requirements-dev.in
-prompt-toolkit==3.0.5     # via ipython
-ptyprocess==0.6.0         # via pexpect
-py==1.8.1                 # via pytest, tox
-pycodestyle==2.6.0        # via flake8
-pycurl==7.43.0.5          # via -r requirements-dev.in
-pyflakes==2.2.0           # via flake8
-pygments==2.6.1           # via ipython
-pyparsing==2.4.7          # via packaging
-pytest==5.4.3             # via -r requirements-dev.in
-pyyaml==5.3.1             # via aspy.yaml, pre-commit
-regex==2020.5.14          # via black
-six==1.15.0               # via packaging, pip-tools, pre-commit, tox, traitlets, virtualenv
-toml==0.10.1              # via black, pre-commit, tox
-tox==3.11.1               # via -r requirements-dev.in
-traitlets==4.3.3          # via ipython
-typed-ast==1.4.1          # via black, mypy
-typing-extensions==3.7.4.2  # via mypy
-virtualenv==20.0.21       # via pre-commit, tox
-wcwidth==0.2.3            # via prompt-toolkit, pytest
-zipp==3.1.0               # via importlib-metadata
+appdirs==1.4.4
+aspy.yaml==1.3.0
+atomicwrites==1.4.0
+attrs==21.2.0
+backports.functools-lru-cache==1.6.4
+cfgv==2.0.1
+click==7.1.2
+configparser==4.0.2
+contextlib2==0.6.0.post1
+distlib==0.3.1
+filelock==3.0.12
+funcsigs==1.0.2
+futures==3.3.0 ; python_version < "3.0"
+identify==1.6.2
+importlib-metadata==2.1.1
+importlib-resources==3.3.1
+mock==3.0.5 ; python_version < "3.0"
+more-itertools==5.0.0
+nodeenv==1.6.0
+packaging==20.9
+pathlib2==2.3.5
+pip-tools==5.5.0
+pluggy==0.13.1
+pre-commit==1.21.0
+py==1.10.0
+pycurl==7.43.0.5
+pyparsing==2.4.7
+pytest==4.6.11
+pyyaml==5.4.1
+scandir==1.10.0
+singledispatch==3.6.1
+six==1.16.0
+toml==0.10.2
+tox==3.23.1
+typing==3.10.0.0 ; python_version < "3.0"
```

Comment:
Don't you need to specify the requirement alternatives for 3.x ?

Reply:
I tested this in both python2.7 and python3.7 environments and it's able to install all dependencies correctly as it is. We may bump some of these packages if we think they work better that way. But since these packages are only meant for develop and are not going to be installed on `pip install` I don't think it's worth it unless we have some issue with a particular version.